pkg/util/strategicpatch/patch.go

```diff
@@ -31,7 +31,352 @@ import (
 // lists should be merged or replaced.
 //
 // For more information, see the PATCH section of docs/api-conventions.md.
+
+// CreateStrategicMergePatch creates a patch that can be passed to StrategicMergePatch.
```

Comment:
Actually, that's almost correct. The current approach can detect changes to the server object, since it computes the divergence in the live state from the original. However, it chooses to fail the update with a conflict if the live state contains changes that are not consistent with the local object.

Without this conflict detection, there isn't much point in an automated process ever making changes to the live state that deviate from the annotation, since they'll simply be discarded the next time the user runs kubectl apply.

Of course, that approach begs a question of timing. How often should the user run kubectl apply to ensure that the live state still conforms to their local configuration? Once an hour? Once a minute? Once a second?

Whatever the answer, it defines a window during which the user willing lets the configuration deviate from their intent. Perhaps they should watch for changes, and quickly update the object if they detect any divergences?

Of course, I'm being facetious.

If what we really is for the configuration to never diverge from what the user specified, then we should simply prevent callers from making conflicting changes in the first place. The annotation contains the configuration requested by the user, and the state is never allowed to deviate from it, other than additions. This policy can be easily enforced on the server side, and eliminates the need for kubectl apply.